ci-operator/step-registry/aws/provision/public-to-private/aws-provision-public-to-private-commands.sh

```diff
@@ -0,0 +1,162 @@
+    local CMD="$1"
+    echo "Running command: ${CMD}"
+    eval "${CMD}"
+    echo ""
+}
+
+function check_cpms_status() {
+    DesiredCount=$(oc get controlplanemachineset/cluster -n openshift-machine-api -o=jsonpath='{.spec.replicas}')
+    COUNTER=0
+    #The rolling update and re-creation of the three masters will take about 1 hour
+    while [ $COUNTER -lt 4500 ]
+    do
+        sleep 300
+        COUNTER=`expr $COUNTER + 300`
+        echo "waiting ${COUNTER}s"
+	#During master rolling update, the API server may not response
+	readyCPMSCount=$(oc get controlplanemachineset/cluster -n openshift-machine-api -o=jsonpath='{.status.readyReplicas}' || true)
+	updatedCPMSCount=$(oc get controlplanemachineset/cluster -n openshift-machine-api -o=jsonpath='{.status.updatedReplicas}' || true)
+        if [[ ${readyCPMSCount} == "${DesiredCount}" ]] && [[ ${updatedCPMSCount} == "${DesiredCount}" ]]; then
+            echo "CPMS update finished."
+            break
+        fi
+    done
+    if [[ ${readyCPMSCount} != "${DesiredCount}" ]] || [[ ${updatedCPMSCount} != "${DesiredCount}" ]]; then
+	echo "Something wrong in the CPMS update:"
+        run_command "oc get machines -n openshift-machine-api"
+        run_command "oc get controlplanemachineset/cluster -n openshift-machine-api -o yaml"
+        return 1
+    fi
+}
+
+
+#Setting the Ingress Controller to private
+echo "Configuring DNS records to be published in the private zone"
+run_command "oc patch dnses.config.openshift.io/cluster --type=merge --patch='{\"spec\": {\"publicZone\": null}}'"
+
+echo "Setting the Ingress Controller to private"
+oc replace --force --wait --filename - <<EOF
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  namespace: openshift-ingress-operator
+  name: default
+spec:
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: Internal
+EOF
+
+
+# Remove the external load balancers from CPMS
+CPMS_PATCH=$(mktemp)
+cat >"${CPMS_PATCH}" <<EOF
+spec:
+  template:
+    machines_v1beta1_machine_openshift_io:
+      spec:
+        providerSpec:
+          value:
+            loadBalancers:
+            - name: ${INFRA_ID}-int
+              type: network
+EOF
+
+echo "Remove the external load balancers from CPMS"
+run_command "oc patch controlplanemachineset/cluster -n openshift-machine-api --type merge --patch-file ${CPMS_PATCH}"
+
+#Wait for CPMS update finished
+#masters was removed from the external LB, need to export proxy to access the cluster
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+else
+  echo "No proxy-conf found, exit now"
+  exit 1
+fi
+
+check_cpms_status
+
+
+#Restricting the API server to private by removing the external LB
+EXT_LB_ARN=$(aws --region ${REGION} elbv2 describe-load-balancers |jq --arg var "${INFRA_ID}" -r '.LoadBalancers[] | select(.LoadBalancerName == "\($var)-ext") | .LoadBalancerArn')
```

Comment:
Shall we also need to remove ingress external lb ?

Reply:
The ingress external LB would be deleted in the step "Setting the Ingress Controller to private", after that command, the Ingress operator will delete the ingress internal LB and create a new internal one, also along with the *.apps Route53 record, the *.apps DNS record will be removed from the public zone and created with a new one to point to the new internal LB